packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js

```diff
@@ -16,26 +16,45 @@ export default class TerminateServiceCtrl {
     return SPECIAL_CONDITIONS_SUBSIDIARIES.includes(this.user.ovhSubsidiary);
   }
 
-  terminate() {
-    this.loading = true;
+  terminateService() {
     return this.confirmTermination(this.service)
       .then(() => this.BillingTerminate.answerForm(this.service, this.model))
-      .then(() =>
-        this.goBack
-          ? this.goBack(true)
-          : this.Alerter.success(
-              this.$translate.instant('billing_confirm_termination_success'),
-            ),
-      )
-      .catch((error) =>
-        this.Alerter.error(
-          this.$translate.instant('billing_confirm_termination_error', {
-            message: error.message,
-          }),
-        ),
-      )
+      .then(() => this.success())
+      .catch((err) => this.error(err))
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  terminateByoipService() {
+    return this.BillingTerminate.terminateByoipService(this.serviceId)
+      .then(() => this.success())
+      .catch((err) => this.error(err))
       .finally(() => {
         this.loading = false;
       });
   }
+
+  terminate() {
+    this.loading = true;
```

Comment:
**nitpick:** setting the `loading` flag here and unsetting it in different functions causes confusions and it is not readable.

Reply:
taken care.